This commit brings in changes to allow building glb-director, glb-hea…

ravisinghsfbay opened the pull request:
…lthcheck, glb-redirect as individual components on a Linux device where all the underlying dependencies are present.

Without these changes, the foregoing components could not individually be built (without the use of Docker) due to mismatches between expectations of different respective versions of gcc and linker.

ravisinghsfbay committed:
This commit brings in changes to allow building glb-director, glb-healthcheck, glb-redirect as individual components on a Linux device where all the underlying dependencies are present.
Without these changes, the foregoing components could not individually be built (without the use of Docker) due to mismatches between expectations of gcc and linker behavior.

brian-pane commented on src/glb-director/util.h:
```diff
@@ -33,7 +33,22 @@
+#ifdef RTE_USE_LIBBSD
 #include "strlcpy.h"
+#else
+
```
This `#else` clause seems unnecessary.

ravisinghsfbay replied:
Agreed. Will remove.

ravisinghsfbay committed:
Code-review comment addressed.
Brought over another change in regards to building glb-redirect from top level when built with target "all"

robloxrob commented:
Does this diverge us from the OSS repo? If it does do we care or do we want to keep in sync with them?

brian-pane commented:
I believe the plan is to contribute this as a PR to the OSS repo.

brian-pane commented:
For the general issue of how to use this repo, my thinking is that everything we commit to this fork should be submitted to the upstream OSS repo so we stay in sync.

robloxrob commented:
Makes sense. We may need to hard fork stuff, but we should try and contribute back as much as possible to make the world a better traffic place.
